theano/tensor/tests/test_basic.py

```diff
@@ -8172,6 +8172,107 @@ def test_norm():
     assert np.allclose(f([1, 1]), np.sqrt(2))
 
 
+class test_cov(unittest.TestCase):
+
+    def test_core(self):
+        x = theano.tensor.matrix('x')
+        c = theano.tensor.cov(x)
+        f = theano.function([x], c)
+
+        # basic cov function
+        data = np.asarray(np.random.rand(3, 5), dtype=config.floatX)
```

Comment:
This series of test could be written more concisely as a loop over shapes. Same for other tests.